kubespawner/utils.py

```diff
@@ -202,3 +202,26 @@ def host_matching(host: str, wildcard: str) -> bool:
     # user.example.com !~ *.user.example.com
     # user.example.com !~ *.example
     return host_parts[1:] == wildcard_parts[1:]
+
+
+# From https://github.com/jupyter-server/jupyter_server/blob/fc0ac3236fdd92778ea765db6e8982212c8389ee/jupyter_server/config_manager.py#L14
+def recursive_update(target, new):
+    """
+    Recursively update one dictionary in-place using another.
+
+    None values will delete their keys.
+    """
+    for k, v in new.items():
+        if isinstance(v, dict):
+            if k not in target:
+                target[k] = {}
+            recursive_update(target[k], v)
+            if not target[k]:
+                # Prune empty subdicts
```

Comment:
Is this definitely the right behaviour? Does K8s consistently treat an absent key as `{}` as opposed to null/None?

Reply:
@manics I've simplified and removed that behavior.